### PR TITLE
added encoding presets and license

### LINKED ISSUE #3: no conversion capability
please add a flag to the docker run so that the webm to mp4 conversion can be disabled

### LINKED ISSUE #5: Copyright issues
SUT has successfully closed Aparat channels that distributed recorded classes. You might want to add personal use only description to the readme.md file and disclaim any abuse made by this code as you might be susceptible to DCMA takedowns.

### LINKED ISSUE #6: Faster encoding
**Fast encode but lower quality medium size**
`ffmpeg.input('video.webm').output(
        'video1.mp4',
        **{'vcodec': 'libx264', 'acodec': 'aac', },
	crf='28',
	preset='ultrafast',
    s='852x480',
	movflags='+faststart'
    ).run()`
**Slow encode but higher quality and lower size**
`ffmpeg.input('video.webm').output(
        'video1.mp4',
        **{'vcodec': 'libx264', 'acodec': 'aac', },
	crf='36',
    tune='stillimage',
	preset='slow',
	movflags='+faststart'
    ).run()
`

### PR DESCRIPTION
**Pull request by sfmth**
I added 3 encoding presets and defaulted it to no encoding as the original webm file is quite good and for long recordings there could be a huge time spent on encoding plus some loss in quality. It's Friday and I couldn't fully test it, you might want to do that.

Fixes #6 
Fixes #5
Fixes #3

### CODE REVIEW

File: src/main.py [hunk 1]
```diff
@@ -348,10 +351,38 @@ def main():
     new_webm_file = os.path.join(download_path, 'video.webm')
     os.rename(webm_file, new_webm_file)
 
-    ffmpeg.input(new_webm_file).output(
-        os.path.join(download_path, 'video.mp4'),
-        **{'vcodec': 'h264', 'acodec': 'mp3', }
-    ).run()
+    if args.encoding == "quality-optimized":
```
Comment (ATofighi) [inline src/main.py:355]:
```suggestion
    ffmpeg_pipe = ffmpeg.input(new_webm_file)
    if args.encoding == "quality-optimized":
```

File: src/main.py [hunk 2]
```diff
@@ -348,10 +351,38 @@ def main():
...
+        ffmpeg.input(new_webm_file).output(
```
Comment (ATofighi) [inline src/main.py]:
```suggestion
        ffmpeg_pipe.output(
```

File: src/main.py [hunk 3]
```diff
@@ -348,10 +351,38 @@ def main():
...
+             os.path.join(download_path, 'video.mp4'),
+             **{'vcodec': 'libx264', 'acodec': 'aac', }, 
+             crf='36', 
+             tune='stillimage', 
+             preset='slow', 
+             movflags='+faststart' 
+        ).run()
+    elif args.encoding == "speed-optimized":
+        ffmpeg.input(new_webm_file).output(
```
Comment (ATofighi) [inline src/main.py]:
```suggestion
        ffmpeg_pipe.output(
```

File: src/main.py [hunk 4]
```diff
@@ -348,10 +351,38 @@ def main():
...
+             os.path.join(download_path, 'video.mp4'), 
+             **{'vcodec': 'libx264', 'acodec': 'aac', }, 
+             crf='28', 
+             preset='ultrafast', 
+             s='852x480', 
+             movflags='+faststart' 
+        ).run()
+    elif args.encoding == "size-optimized":
+        ffmpeg.input('video.webm').output(
```
Comment (ATofighi) [inline src/main.py]:
```suggestion
        ffmpeg_pipe.output(
```

**Review by ATofighi — CHANGES_REQUESTED** [review]
Thanks for you contribution. 

Looks good to me.

I only suggest a change that reuses `ffmpeg.input(donwloaded_file)` and doesn't repeat it.

+ fix flake8 issue that occur for too many new lines at line 384 and 385.  [https://travis-ci.com/github/ATofighi/skyroom-record/builds/219153531](url).

**Comment by ATofighi** [comment]
Fixes #6 #5 #3

**Comment by sfmth** [comment]
Alright all set I guess

**Review by ATofighi — APPROVED** [review]

**Pull request merged**